src/main/java/org/wso2/carbon/module/swiftiso20022/JsonToMT940Transformer.java

```diff
@@ -56,8 +54,8 @@ public void connect(MessageContext messageContext) throws ConnectException {
             if (payload.isPresent()) {
                 RequestPayloadModel requestPayload = gson.fromJson(payload.get(), RequestPayloadModel.class);
 
-                ErrorModel validationResponse = validateRequestPayload(requestPayload);
-                if (validationResponse.isError()) {
+                ValidationResult validationResponse = validateRequestPayload(payload.get());
```

Comment:
Also shall we consider renaming the `payload` to `optionalPayload`?

```suggestion
                ValidationResult validationResult = validateRequestPayload(payload.get());
```

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/6/commits/8076e9b5033a9608b6839de38320ec994e01055d